elastic/kibana
# [Security Solutions] Fixes telemetry to work with rule types

## Summary

What this does:
* Fixes telemetry to work with the newer rule types
* Updates the queries to the new rule types and rule query names
* Uses constants where I can from cases and the new rule types
* Changes the index to the new index type alias
* Adds e2e backend tests we didn't have before

What this does not do:
* Doesn't add e2e backend tests for alerts added to cases
* Doesn't add e2e backend tests for ML jobs for security_solution

Those two test scenarios have to be manually tested still.

Manual testing:
To see telemetry go to advanced settings -> Usage Data (and click cluster data):
<img width="2193" alt="Screen Shot 2021-12-08 at 4 14 43 PM" src="https://user-images.githubusercontent.com/1151048/145310671-b7350892-d290-4f3d-ab8c-4e9ec86f4120.png">

Create alerts of different types and add them to cases:
<img width="1464" alt="Screen Shot 2021-12-08 at 4 48 21 PM" src="https://user-images.githubusercontent.com/1151048/145310800-2fae6373-5e84-46ec-9e44-f7a140ea9c36.png">

Activate ML_jobs and any alerts that have ML jobs associated:
<img width="754" alt="Screen Shot 2021-12-08 at 5 08 42 PM" src="https://user-images.githubusercontent.com/1151048/145310978-861f4bb7-2575-4a07-a55f-1e4fdfe288e7.png">

When clicking advanced settings -> Usage Data -> Click cluster data

Search for `security_solution` and then ensure that the data looks as expected underneath the different values such as:

`ml_jobs`
<img width="750" alt="Screen Shot 2021-12-08 at 3 08 25 PM" src="https://user-images.githubusercontent.com/1151048/145311124-c3523d4e-b31b-4bab-b14e-267155bf2b92.png">

`detection_rules` and `cases` working again:
<img width="420" alt="Screen Shot 2021-12-08 at 4 43 10 PM" src="https://user-images.githubusercontent.com/1151048/145311192-e062c435-e8c3-4919-b4e9-8a786dc588c6.png">

Note, `detection_rule_detail` will only be filled in if have prepackaged rules installed:
<img width="761" alt="Screen Shot 2021-12-08 at 5 14 50 PM" src="https://user-images.githubusercontent.com/1151048/145311446-1d78541f-1211-4389-b947-7c0939d7c946.png">

Also note that the `detection_rule_detail`'s `rule_id` is its UUID and not its `rule_id`. That's the way it's been in the codebase for a while it looks like so I have not changed that behavior.

### Checklist
